src/Ewald_ParticleMesh.cpp

```diff
@@ -265,7 +344,175 @@ int Ewald_ParticleMesh::CalcNonbondEnergy(Frame const& frameIn, AtomMask const&
   }
 
   e_vdw = 0.0;
-  double e_direct = Direct( pairList_, e_vdw );
+  double e_direct = Direct( pairList_, e_vdw);
+  if (debug_ > 0)
+    mprintf("DEBUG: Eself= %20.10f   Erecip= %20.10f   Edirect= %20.10f  Evdw= %20.10f\n",
+            e_self, e_recip, e_direct, e_vdw);
+  e_vdw += (e_vdw_lr_correction + e_vdw6self + e_vdw6recip);
+  t_total_.Stop();
+  e_elec = e_self + e_recip + e_direct;
+  return 0;
+}
+
+/** Calculate full nonbonded energy with PME Used for GIST, adding 6 arrays to store the
+ * decomposed energy terms for every atom
+ */
+int Ewald_ParticleMesh::CalcNonbondEnergy_GIST(Frame const& frameIn, AtomMask const& maskIn,
+                                      double& e_elec, double& e_vdw,
+                                      std::vector<double>& e_vdw_direct,
+                                      std::vector<double>& e_vdw_self,
+                                      std::vector<double>& e_vdw_recip,
+                                      std::vector<double>& e_vdw_lr_cor,
+                                      std::vector<double>& e_elec_self,
+                                      std::vector<double>& e_elec_direct,
+                                      std::vector<double>& e_elec_recip,
+                                      std::vector<int>& atom_voxel )
+{
+  t_total_.Start();
+  Matrix_3x3 ucell, recip;
+  double volume = frameIn.BoxCrd().ToRecip(ucell, recip);
+
+  auto step0 = std::chrono::system_clock::now();
+
+
+  double e_self = Self_GIST( volume, e_elec_self ); // decomposed E_elec_self for GIST
+
+  auto step1 = std::chrono::system_clock::now();
```

Comment:
Because cpptraj is distributed with Amber, C++11 stuff still has to be optional. All C++11 stuff should be hidden behind the define `C11_SUPPORT`